lib/tasks/config_files.rake

```diff
@@ -144,6 +144,21 @@ namespace :config_files do
     end
   end
 
+  desc 'Unset reject_mail_at_mta on a request'
+  task :unset_reject_mail_at_mta => :environment do
```

Comment:
Yeah, doing anything else would add a load of complexity.

I did start thinking about this. Here's a bit of a mind dump of my thought process when I looked in to it.

Feel free to completely ignore this, but I guess its worth posting if we're ever wondering why we did it this way.

---

Lets say we use `:unset_reject_mail_at_mta` to set `:reject_mail_at_mta` to `false`on a request that's rejecting responses at MTA level. Running `config_files:set_reject_mail_at_mta` is going to reset this to `true` again the next time its run, ergo `config_files:unset_reject_mail_at_mta` will basically get ignored. 

I don't think I have a _great_ idea of how to solve this, but here are some options.

I'll add specific comments that I noted when I was going through the commits, but they might be ignorable depending on how we deal with this situation.

I think we're trying to handle three situations:

- Not rejecting at MTA because its under the threshold
- Rejecting at MTA
- Not rejecting at MTA because we explicitly said not to

My initial thought was to invert the logic of `reject_mail_at_mta`; make everything `true` by default ("fair game for rejection"), generate the list based on requests that are `true` and over the threshold and exclude requests that are explicitly marked as `false`.

```ruby
# By default, anything is fair game to get rejected at MTA level
# Actually want this done in the migration, but YSWIM
InfoRequest.update_all(:reject_mail_at_mta => true)

# Have a query that finds all that are over thereshold and "fair game"
class InfoRequest
  def self.over_rejection_at_mta_threshold(options)
    where(["updated_at < (now() -
            interval '#{options[:age_in_months]} months')
            AND allow_new_responses_from = 'nobody'
            AND rejected_incoming_count >= ?
            AND reject_mail_at_mta = 't'
            AND url_title <> 'holding_pen'",
            options[:rejection_threshold]])
  end

#
# Rake Tasks
#

# Generate a list of any requests over the threshold and OK to be rejected
task :generate_mta_rejection_list => :environment do
  options = { :rejection_threshold => ENV['REJECTED_THRESHOLD'],
              :age_in_months => ENV['AGE_IN_MONTHS'],
              :dryrun => dryrun }
  puts InfoRequest.
         over_rejection_at_mta_threshold(options).
           pluck(:incoming_email)
end

# Manually stop mail getting rejected for a given request
# The `false` option is a manual marker to stop it getting
# included in the rejection list, unless explicitly changed back
# to `true`
task :unset_reject_mail_at_mta => :environment do
  info_request = InfoRequest.find(ENV['REQUEST_ID'])
  info_request.update_attributes!(:reject_mail_at_mta => false)
end
```

However, doing it this way around makes it trickier to handle the UI elements, and installs who don't configure MTA rejection. i.e. what happens here?

```erb
<% if @info_request.reject_mail_at_mta? %>
  <p><strong>Mail for this request is automatically being rejected at the mail server due to spam.</strong></p>
<% else %>
```

I guess the dumb way of doing this is having a config flag and some instance method like:

```erb
<% if AlaveteliConfiguration.reject_at_mta && @info_request.over_rejection_threshold? %>
  <p><strong>Mail for this request is automatically being rejected at the mail server due to spam.</strong></p>
<% else %>
  …
<% end %>
```

```ruby
def over_rejection_threshold?(options = {})
  self.class.over_rejection_at_mta_threshold(options).
    where(:id => id).
      any?
end
```

Downsides to this are:

- "is being rejected" logic is much more confusing
- Would probably end up having to add config flags for the thresholds, given that we don't know what good defaults are yet
- The current initiation of MTA rejection is simpler – just run the rake task

Another alternative is not setting a default value for `reject_at_mta`.

| `reject_at_mta` value | Implication |
| -------------------|----------|
| `nil` | "fair game" for rejection |
| `true` | Is being rejected |
| `false` | Do not reject at MTA |

This allows `InfoRequest#reject_at_mta?` to behave in a similar way to how it does now, and we'd just need to change the query to use `AND reject_mail_at_mta IS NULL`.

I'm not sure whether I like the idea of a boolean field allowing a `NULL` value, but it does sort of model what we're trying to say. I thought about using a `String` field instead, but what would the values be? `%q(can_be_rejected yes no)`? Meh.

---

So yeah, that's about the point where I just figured you can change `allow_new_responses_from` in `unset_reject_mail_at_mta` and make the whole thing a lot less of a faff.